mxcube3/routes/qutils.py

```diff
@@ -1431,23 +1442,36 @@ def queue_model_child_added(parent, child):
             parent_entry.enqueue(dcg_entry)
 
 
-def queue_model_diff_plan_available(char, index, collection):
-    if isinstance(collection, qmo.DataCollection):
-        if collection.get_origin():
-            origin_model, origin_entry = get_entry(collection.get_origin())
-        collection.set_enabled(False)
-        dcg_model = char.get_parent()
-        sample = dcg_model.get_parent()
-        task = _handle_dc(sample._node_id, collection)
-        task.update({'isDiffractionPlan': True, 'originID': origin_model._node_id})
-        socketio.emit('add_diff_plan', {"tasks": [task]}, namespace='/hwr')
+def queue_model_diff_plan_available(char, index, collection_list):
+    # TODO: The client is not prepared for handling several collections for now @#@#!
```

Comment:
I don't really understand what you mean with that the queue is not prepared for handling several collections ? What's missing ?

Reply:
For the `case auto_add_diff_plan = false`, we allow the user to review the diff plan by opening a standard data collection modal. If the diff plan contains more than one collection, we still display the same modal and I think show only the first collection (I am not fully sure about which one though). We discussed this on my last visit, we said that for now it is ok like this.

What is missing is to design a new modal that could display several collections from the diff plan in a fancy way.

Reply:
Aha, I see. 

I agree that its nice to have a dialog where we can change the parameters of multiple collections at once. However I believe, unless you have any idea on how to do that right now, that it something that requires some thought and time to implement. Its not a very important feature so we should probably wait for this until maybe 3.1 or an intermediate release 3.0.x.

Id rather that we use what we have today in-order to achieve what we want. So Id suggest, the user clicks add diffraction plan, and gets no dialog, all collections are added to the queue. The user can then remove or change the parameters of any of those like she would with any other task. 

There are of course alternatives to this, like the user could select which collections to add and so on. Or we only display the dialog when there is one collection on no dialog when there are multiple.

My interpretation was that it was okey for the moment as development in progress bu not to give to users. In anyways sorry for any misunderstanding.

Characterisations is something that are used alot so the behavior needs to be consistent and convenient. So I see two solutions for 3.0 we only allow diffraction plans with on collection in them, thus what you've made today is basically fine. The second solution, which I prefer, is something along with what I described above (we only need to remove the dialog).

Marcus

Reply:
I kind of agree with you. In fact, I remember that we said that if more than one collection, add all collections without user intervention.

But, surprise surprise... I had completely forgoten but we already have that implemented!
https://github.com/mxcube/mxcube3/blob/master/mxcube3/ui/components/SampleQueue/CharacterisationTaskItem.jsx#L71

I do not know if it is working...

Reply:
Okey, so it remains to be tested :)

Reply:
I will try to test with mockups

Reply:
working with the last fix